# @khulnasoft/y

TODO: Description

## Rule Details

### Examples of **incorrect** code for this rule

TODO

### Examples of **correct** code for this rule

TODO

## Options

Nothing
TODO: or describe here

## Related rules

- TBA

## When Not To Use It

TODO
